client/components/Recordings.jsx

```diff
@@ -26,60 +34,115 @@ export default class Recordings extends React.Component {
     };
   }
 
+  titleName(e) {
+    this.setState({ title: e.target.value });
+  }
+
   startRecording(e) {
     e.preventDefault();
+    this.setState({ recordings: [] });
     this.chunks = [];
     this.mediaRecorder.start(10);
-    this.setState({ recording: true });
+    this.setState({ recording: true, formInputs: false, duration: 0 }, () => {
+      this.time = setInterval(() => this.setState({ duration: this.state.duration + 1 }), 1000);
+    });
   }
 
   stopRecording(e) {
     e.preventDefault();
     this.mediaRecorder.stop();
-    this.setState({ recording: false });
+    clearInterval(this.time);
+    this.setState({ recording: false, formInputs: true });
     this.saveAudio();
   }
 
   saveAudio() {
     const blob = new Blob(this.chunks, { type: 'audio/webm' });
     const audioURL = URL.createObjectURL(blob);
-    const recordings = this.state.recordings.concat([audioURL]);
-    this.setState({ recordings });
+    this.file = new File([blob], `${blob}.webm`, { type: 'audio/webm' });
```

Comment:
I get the feeling that this is creating a massive string. for your file name. how about `Date.now()` instead of the whole blob? Would that break anything?